packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTableSet.tsx

```diff
@@ -0,0 +1,286 @@
+    const preResults: Array<IApiProperty> = [];
+
+    const { jsonDocs } = this.props;
+
+    if (jsonDocs) {
+      const propsName: string = `I${jsonDocs.name}Props`;
+      for (let j = 0; j < jsonDocs.tables.length; j++) {
+        switch (jsonDocs.tables[j].kind) {
+          case 'enum': {
+            const enumMembers: IApiEnumProperty[] = [];
+
+            const members: IEnumTableRowJson[] = jsonDocs.tables[j].members as IEnumTableRowJson[];
+            for (let k = 0; k < members.length; k++) {
+              // each member within the enum
+              enumMembers.push({
+                description: members[k].descriptionHtml,
+                name: members[k].name,
+                value: members[k].value
+              });
+            }
+
+            // the enum
+            preResults.push({
+              propertyName: jsonDocs.tables[j].name,
+              description: jsonDocs.tables[j].descriptionHtml,
+              title: jsonDocs.tables[j].kind ? jsonDocs.tables[j].name + ' ' + jsonDocs.tables[j].kind : jsonDocs.tables[j].name,
+              propertyType: PropertyType.enum,
+              property: enumMembers
+            });
+
+            break;
+          }
+          case 'interface': {
+            const interfaceMembers: IApiInterfaceProperty[] = [];
+
+            const members: ITableRowJson[] = jsonDocs.tables[j].members as ITableRowJson[];
+            for (let k = 0; k < members.length; k++) {
+              // each member within the interface or class
+              interfaceMembers.push({
+                description: members[k].descriptionHtml,
+                name: members[k].name,
+                typeTokens: members[k].typeTokens,
+                deprecated: members[k].deprecated,
+                defaultValue: members[k].defaultValue || ''
+              });
+            }
+
+            // the interface
+            if (propsName === jsonDocs.tables[j].name) {
+              iComponentProps.push({
+                propertyName: jsonDocs.tables[j].name,
+                description: jsonDocs.tables[j].descriptionHtml,
+                extendsTokens: jsonDocs.tables[j].extendsTokens,
+                title: jsonDocs.tables[j].kind ? jsonDocs.tables[j].name + ' ' + jsonDocs.tables[j].kind : jsonDocs.tables[j].name,
+                propertyType: PropertyType.interface,
+                property: interfaceMembers
+              });
+            } else {
+              preResults.push({
+                propertyName: jsonDocs.tables[j].name,
+                description: jsonDocs.tables[j].descriptionHtml,
+                extendsTokens: jsonDocs.tables[j].extendsTokens,
+                title: jsonDocs.tables[j].kind ? jsonDocs.tables[j].name + ' ' + jsonDocs.tables[j].kind : jsonDocs.tables[j].name,
+                propertyType: PropertyType.interface,
+                property: interfaceMembers
+              });
+            }
+
+            break;
+          }
+          case 'class': {
```

Comment:
You might consider refactoring these blocks of code into individual methods for maintainability. There seems to be a bit of logic that is common across the various cases.